server/hotrod/src/main/java/org/infinispan/server/hotrod/KeyValueRetrieverConverter.java

```diff
@@ -0,0 +1,86 @@
+package org.infinispan.server.hotrod;
+
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Set;
+
+import org.infinispan.commons.io.UnsignedNumeric;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.container.versioning.NumericVersion;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.notifications.cachelistener.filter.CacheEventConverter;
+import org.infinispan.notifications.cachelistener.filter.EventType;
+
+class KeyValueRetrieverConverter implements CacheEventConverter<byte[], byte[], byte[]> {
+   private KeyValueRetrieverConverter() {
+   }
+
+   public static KeyValueRetrieverConverter SINGLETON = new KeyValueRetrieverConverter();
+
+   @Override
+   public byte[] convert(byte[] key, byte[] oldValue, Metadata oldMetadata, byte[] newValue, Metadata newMetadata, EventType eventType) {
+      int capacity = UnsignedNumeric.sizeUnsignedInt(key.length) + key.length +
+            (newValue != null ? UnsignedNumeric.sizeUnsignedInt(newValue.length) + newValue.length + 8 : 0) +
+            (oldValue != null ? UnsignedNumeric.sizeUnsignedInt(oldValue.length) + oldValue.length + 8 : 0);
+
+      byte[] out = new byte[capacity];
+      int offset = UnsignedNumeric.writeUnsignedInt(out, 0, key.length);
+      offset += putBytes(key, offset, out);
+      if (newValue != null) {
+         offset += UnsignedNumeric.writeUnsignedInt(out, offset, newValue.length);
+         offset += putBytes(newValue, offset, out);
+         putLong(((NumericVersion) newMetadata.version()).getVersion(), offset, out);
+      }
+      if (oldValue != null) {
```

Comment:
Hrmm, I am debating if we want to only just return a single value instead of both. I know this only affects modification events, which you aren't currently using.

Reply:
this means we leave it as it is ?

Reply:
so this class has been removed, we use the same class for both.

Reply:
Ok, as discussed elsewhere we shouldn't modify the format of the existing converter so we'll include either the new value or the old value, never both.